pkg/autoscaler/autoscaler/pdplan_autoscaler.go

```diff
@@ -128,6 +131,108 @@ func (am *autoScalerManager) updateAutoscalingClusters(tac *v1alpha1.TidbCluster
 }
 
 func (am *autoScalerManager) createAutoscalingClusters(tc *v1alpha1.TidbCluster, tac *v1alpha1.TidbClusterAutoScaler, groupsToCreate []string, groupPlanMap map[string]pdapi.Plan) error {
-	// TODO in next PR
-	return nil
+	var errs []error
+	for _, group := range groupsToCreate {
+		plan := groupPlanMap[group]
+		component := plan.Component
+
+		if !checkAutoscalingComponent(tac, component) {
+			continue
+		}
+
+		resource, ok := tac.Spec.Resources[plan.ResourceType]
+		if !ok {
+			errs = append(errs, fmt.Errorf("unknown resource type %v for group %s tac[%s/%s]", plan.ResourceType, plan.Labels[groupLabelKey], tac.Namespace, tac.Name))
+			continue
+		}
+		resList := corev1.ResourceList{
+			corev1.ResourceCPU:    resource.CPU,
+			corev1.ResourceMemory: resource.Memory,
+		}
+
+		autoTcName, err := genAutoClusterName(tac, component, plan.Labels, resource)
+		if err != nil {
+			errs = append(errs, err)
+			continue
+		}
+
+		autoTc := &v1alpha1.TidbCluster{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      autoTcName,
+				Namespace: tc.Namespace,
+				OwnerReferences: []metav1.OwnerReference{
+					controller.GetTiDBClusterAutoscalerOwnerRef(tac),
+				},
+			},
+			Status: v1alpha1.TidbClusterStatus{
+				AutoScaler: &v1alpha1.TidbClusterAutoScalerRef{
+					Name:      tac.Name,
+					Namespace: tac.Namespace,
+				},
+			},
+			Spec: v1alpha1.TidbClusterSpec{
+				Cluster: &v1alpha1.TidbClusterRef{
+					Name:      tc.Name,
+					Namespace: tc.Namespace,
+				},
+			},
+		}
+
+		switch component {
+		case v1alpha1.TiKVMemberType.String():
+			resList[corev1.ResourceStorage] = resource.Storage
+
+			autoTc.Spec.TiKV = tc.Spec.TiKV.DeepCopy()
+			autoTc.Spec.TiKV.Replicas = int32(plan.Count)
+			autoTc.Spec.TiKV.ResourceRequirements = corev1.ResourceRequirements{
+				Limits:   resList,
+				Requests: resList,
+			}
+			if autoTc.Spec.TiKV.Config == nil {
```

Comment:
Should the relationship of the following condition be:
```
if autoTc.Spec.TiKV.Config == nil
else if autoTc.Spec.TiKV.Config.Server == nil
else if autoTc.Spec.TiKV.Config.Server.Labels == nil
```

Reply:
I think the logic here is to initialize the `Config` / `Config.Server` / `Config.Server.Labels` if they are `nil`.

Reply:
Yes, but if the first if is true, the other two will be false. So we don't need them.

Reply:
updated